attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.autoconfigure.health.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Kafka health indicator. Check that the consumer group can be accessed and is registered with the cluster,
+ * if not mark the service as unhealthy.
+ */
+@Component("kafka")
+@ConditionalOnEnabledHealthIndicator("kafka")
+public class KafkaHealthIndicator implements HealthIndicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaHealthIndicator.class);
+
+    @Value("${akka.kafka.consumer.kafka-clients.group.id}")
+    private String groupId;
+
+    private final AdminClient adminClient;
+
+    /**
+     * Requires the AdminClient to interact with Kafka
+     * @param adminClient of the cluster
+     */
+    public KafkaHealthIndicator(final AdminClient adminClient) {
+        this.adminClient = adminClient;
+    }
+
+    @Override
+    public Health getHealth(final boolean includeDetails) {
+        return Optional.of(performCheck())
+                .filter(healthy -> healthy)
+                .map(up -> Health.up().withDetail("group", this.groupId).build())
+                .orElseGet(() -> Health.down().withDetail("group", this.groupId).build());
+    }
+
+    /**
+     * Health endpoint
+     * @return the {@code Health} object
+     */
+    @Override
+    public Health health() {
+        return Optional.of(performCheck())
+                .filter(healthy -> healthy)
+                .map(up -> Health.up().build())
+                .orElseGet(() -> Health.down().build());
+    }
+
+    private boolean performCheck() {
+        try {
+            Map<String, ConsumerGroupDescription> groupDescriptionMap = this.adminClient.describeConsumerGroups(Collections.singletonList(this.groupId))
+                            .all()
+                            .get(1, TimeUnit.SECONDS);
+
+            ConsumerGroupDescription consumerGroupDescription = groupDescriptionMap.get(this.groupId);
+
+            LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());
+
+            if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
+                return consumerGroupDescription.members().stream()
+                        .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            LOGGER.debug("Timeout during Kafka health check for group {}", this.groupId, e);
```

Comment:
If sonarqube complains, it may be appeased by changing this to a higher logging level?
```suggestion
            LOGGER.warn("Timeout during Kafka health check for group {}", this.groupId, e);
```